jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -56,38 +51,98 @@
 {
     private static final Logger LOGGER = Log.getLogger(Pool.class);
 
-    private final List<Entry> sharedList = new CopyOnWriteArrayList<>();
+    private final List<Entry> entries = new CopyOnWriteArrayList<Entry>()
+    {
+        @Override
+        public ListIterator<Entry> listIterator(int index)
+        {
+            try
+            {
+                return super.listIterator(index);
+            }
+            catch (IndexOutOfBoundsException e)
+            {
+                return listIterator();
+            }
+        }
+    };
```

Comment:
Overriding `listIterator(int)` is not necessary anymore, since we never use it, and we don't leak the `entries` field.